client/src/origo_native.rs

```diff
@@ -150,7 +150,14 @@ async fn handle_origo_mode(
   client_socket.close().await?;
 
   // Close the underlying TCP stream.
-  tls_receiver.await?.1.into_inner().into_inner().into_inner().into_inner().0.shutdown().await?;
+  match tls_receiver.await? {
```

Comment:
why is this `shutdown()` call returning a `Result<Result<..>>`? That's horrific lol

Reply:
Seems like it could be flattened, but maybe that's out of our control.

Reply:
i had the exact same reaction when i saw this lol
`compat.into_inner().into_inner().into_inner().into_inner().0.shutdown()`